From 879a4cfcd6b0a31d7f483d91a3685f4c497036cb Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Tue, 4 Nov 2014 00:13:21 +0100
Subject: [PATCH 1/6] linux: Check for Logind first, then *Kit-s with UPower
 when detecting PM framework

Logind is the most likely candidate to work nowadays and latest UPower does
not provide suspend/hibernate API.
---
 xbmc/powermanagement/PowerManager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/powermanagement/PowerManager.cpp b/xbmc/powermanagement/PowerManager.cpp
index 5900f98..27948b5 100644
--- a/xbmc/powermanagement/PowerManager.cpp
+++ b/xbmc/powermanagement/PowerManager.cpp
@@ -75,12 +75,12 @@ void CPowerManager::Initialize()
   m_instance = new CAndroidPowerSyscall();
 #elif defined(TARGET_POSIX)
 #if defined(HAS_DBUS)
-  if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
+  if (CLogindUPowerSyscall::HasLogind())
+    m_instance = new CLogindUPowerSyscall();
+  else if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
     m_instance = new CConsoleUPowerSyscall();
   else if (CConsoleDeviceKitPowerSyscall::HasDeviceConsoleKit())
     m_instance = new CConsoleDeviceKitPowerSyscall();
-  else if (CLogindUPowerSyscall::HasLogind())
-    m_instance = new CLogindUPowerSyscall();
   else if (CUPowerSyscall::HasUPower())
     m_instance = new CUPowerSyscall();
   else
-- 
2.1.1

